### ISSUE TITLE
Testing Harness for the Interpreter

### ISSUE THREAD

**Issue by Lythenas** [sp-uulm/MiniLua#39]
As a developer I want to be able to easily add new tests for the interpreter.

---

Test should/could contain:

- lua source code
- text for stdin
- text that is expected of stdout
- text that is expected of stderr
- if the test is supposed to fail (and maybe with what error or on what line the error is)

Additionally the `assert` function in lua can be used to check things (also needs to be implemented).

Examples:

```lua
-- expected stdout: "Hello, World!"
if true then
  print("Hello, World!")
end
```

```lua
-- fails
"abc" + 5 -- expected failure: can't add values of type string and number
```

This can either all be in the same file (e.g. by putting input and expected values in comments at the top) or split up into multiple files (e.g. `test_name.lua`, `test_name.stdin`, `test_name.stdout`, `test_name.stderr`, `test_name.fail`) or something similar.

If we do this in a single file (which I think is easier as long as we keep the in/outputs relatively small) regex search for the stdin/etc. comments would be easiest to implement.

---

Tasks:

- [ ] Harness to easily run these tests
  - Maybe separate from the C++ tests
    - But could be done e.g. using Catch2s `DYNAMIC_SECTION`.
  - Executable or script that executes all tests and reports the result

**Comment by Lythenas** [comment]
Currently we have simple tests by using `assert` in lua code. We can't test things like `print` or errors like this. But creating a complicated testing harness like this is to much work for now.

**Comment by Lythenas** [comment]
The basis for this was implemented in #126
